### PR TITLE
Support for branches

### PR DESCRIPTION
**Pull request by efirs**
Usage:

```
tigris branch create br1
tigris branch create br1 --checkout
tigris branch delete br1
tigris branch checkout br1
tigris branch list
```

Checkout is per user. Later we can add better granularity, e.g. per
project directory.

### CODE REVIEW

File: cmd/branch.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+			}
+
+			util.Infof("Branch %s deleted successfully", args[0])
+
+			return nil
+		})
+	},
+}
+
+func checkoutBranch(name string) {
```
Comment (ovaistariq) [inline cmd/branch.go:72]: Does this config only apply to the tigris CLI?

Comment (efirs) [reply cmd/branch.go:72]: With this diff, yes.
Later we can extend. PR didn't update commit message:
```
Checkout is per user. Later we can add better granularity, e.g. per
project directory.
```

Comment (efirs) [reply cmd/branch.go:73]: We need to unify config, so all SDKs and CLI read the same set of config files, e.g. /etc/tigris.yaml, ~/.tigris.yaml ~/proj1/.tigris.yaml

Comment (ovaistariq) [reply cmd/branch.go:73]: I think the current approach of reading from environment flags is much better as it works seamlessly across all the modern application deployment environments whether it is k8s, AWS lambda, Vercel, etc.

Comment (ovaistariq) [reply cmd/branch.go:73]: What do you mean checkout is per-user?

Comment (efirs) [reply cmd/branch.go:73]: Right, environment variables should be used in production deployment.
Config files approach is convenience for local development.
User just run `tigris branch checkout` and both CLI and his compiled application use the same branch.

Comment (efirs) [reply cmd/branch.go:73]: /etc/tigris/tigris.yaml (system)
~/tigris/tigris-cli.yaml (per-user) (this is where branch is saved currently)
~/proj1/.tigris.yaml (per project)

Comment (ovaistariq) [reply cmd/branch.go:73]: IMO there should be just one source of truth which the application reads the config from, which is what the environment file is. This is also the convention in the TypeScript world.

Comment (efirs) [reply cmd/branch.go:73]: When we implement per project branch handling, we can save it in the TypeScript .env file, but this is out of scope of this diff.

Comment (ovaistariq) [reply cmd/branch.go:73]: Does the typescript sdk read this config file? If not then the env discussion is unrelated in the context of this diff.

Comment (efirs) [reply cmd/branch.go:73]: It doesn't.

**Review by ovaistariq — APPROVED** [review]

**Pull request merged**